### PR TITLE
dotCMS/core#19133 Can't push publish from old dialog

### PR DESCRIPTION
**Pull request by hmoreras**

### CODE REVIEW

File: src/app/api/services/dot-workflow-event-handler/dot-workflow-event-handler.service.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+                    }
+                });
+        } else {
+            this.openWizard(event);
+        }
+    }
+
+    /**
+     * Check if there are environments present otherwise send a notification
+     * @memberof DotWorkflowEventHandlerService
```
Comment (alfredo-dotcms) [inline src/app/api/services/dot-workflow-event-handler/dot-workflow-event-handler.service.ts:67]: @returns

File: src/app/api/services/dot-workflow-event-handler/dot-workflow-event-handler.service.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+                }
+                return false;
+            })
+        );
+    }
+
+    /**
+     * Check if Push Publish is par of th sub-actions of the workflow.
+     * @param {DotCMSWorkflowInput[]} inputs
+     * returns boolean
```
Comment (alfredo-dotcms) [inline src/app/api/services/dot-workflow-event-handler/dot-workflow-event-handler.service.ts:93]: typo

File: src/app/api/services/dot-workflow-event-handler/dot-workflow-event-handler.service.ts [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        return steps.length
+            ? {
+                  title: title,
+                  steps: steps
+              }
+            : null;
+    }
+
+    /**
+     * conver the data collected to what is expecting the endpoint.
```
Comment (alfredo-dotcms) [inline src/app/api/services/dot-workflow-event-handler/dot-workflow-event-handler.service.ts:127]: typo

File: src/app/api/services/dot-workflow-event-handler/dot-workflow-event-handler.service.ts [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+     * @memberof DotWorkflowEventHandlerService
+     */
+    processWorkflowPayload(
+        data: { [key: string]: any },
+        inputs: DotCMSWorkflowInput[]
+    ): { [key: string]: any } {
+        if (this.containsPushPublish(inputs)) {
+            data['whereToSend'] = data.environment.join();
+            data['iWantTo'] = data.pushActionSelected;
+            data['publishTime'] = moment(data.publishDate).format('HH-mm');
```
Comment (alfredo-dotcms) [inline src/app/api/services/dot-workflow-event-handler/dot-workflow-event-handler.service.ts:140]: check if this throws a warning in console due to it's deprecated, if it does then the fix is
`moment(new Date(data.publishDate)).format('HH-mm');`

File: src/app/portlets/dot-porlet-detail/dot-portlet-detail.component.spec.ts [hunk 5]
```diff
@@ -1,45 +1,106 @@
 /* tslint:disable:no-unused-variable */
```
Comment (alfredo-dotcms) [inline src/app/portlets/dot-porlet-detail/dot-portlet-detail.component.spec.ts:1]: remove

File: src/app/api/services/dot-workflow-event-handler/dot-workflow-event-handler.service.ts (hunk 4 above)
Comment (hmoreras) [reply src/app/api/services/dot-workflow-event-handler/dot-workflow-event-handler.service.ts:146]: no.

**Pull request merged**